src/reducers/actionHelpers/datasource.js

```diff
@@ -165,8 +162,8 @@ export const addNewRow = (state, { rowId, stateKey }) => {
         data: newData,
         proxy: data,
         isEditing: true,
-        lastUpdate: generateLastUpdate(),
-        total: newData.size
+        total: existingState.get('total') + 1,
```

Comment:
Probably need to be a little bit more defensive here sadly. I would add a check to make sure existingState has been initialized, otherwise this will throw